### PR TITLE
err: Implement basis for new error API

### PR DESCRIPTION
**Pull request by Geod24**
Fairly straightforward, didn't cover the full header though.

### CODE REVIEW

File: source/deimos/openssl/err.d [hunk 1]
```diff
@@ -327,7 +312,50 @@ struct ERR_string_data_st {
...
+	auto ERR_raise () (int lib, int reason) {
+		return ERR_raise_data(lib, reason, null);
+	}
+
+	// WARNING: The string passed to this function must be `\0` terminated
+	// This works for literal, but for non-default arguments, beware.
+	// Note that the macro in OpenSSL v3 does not take argument,
+	// but we need to if we want to present the same API.
+	auto ERR_raise_data () (
+		string file = __FILE__, int line = __LINE__, string func = __FUNCTION__)
```
Comment (CyberShadow) [inline source/deimos/openssl/err.d:342]: Signature in [docs](https://www.openssl.org/docs/manmaster/man3/ERR_raise_data.html) seems to be:

    void ERR_raise_data(int lib, int reason, const char *fmt, ...);

**Comment by Geod24** [comment]
Hum I don't think the FILE/LINE/FUNC thing can work with the same API.

**Comment by Geod24** [comment]
@CyberShadow : Updated. Not found of the API, but it's the only one that provides the same functionality (automatic file / line / func). Any suggestion / idea ?

**Review by CyberShadow — APPROVED** [review]
LGTM. I think an alternative which would be closer to the C macros would be to make `ERR_raise_data` a `@property` function which returns `&ERR_set_error`. Then, `ERR_raise_data` wouldn't need to copy `ERR_set_error`'s signature.

**Comment by CyberShadow** [comment]
Right, I forgot properties are useless :(

I don't think you can do better without requiring object files.

**Pull request merged**